File: first_unique_char_in_a_string/first_unique_char_in_a_string.go
package main

import "fmt"

func firstUniqChar(s string) int {
    freq := make(map[rune]int)

    // Count the frequency of each character
    for _, char := range s {
        freq[char]++
    }

    // Find the first unique character
    for i, char := range s {
        if freq[char] == 1 {
            return i
        }
    }

    // Return -1 if no unique character is found
    return -1
}

func main() {
    fmt.Println(firstUniqChar("leetcode")) // Output: 0 (Character 'l' is unique)
}

File: first_unique_char_in_a_string/first_unique_char_in_a_string.exs
defmodule UniqueCharacter do
  def first_uniq_char(s) do
    freq = Enum.reduce(s, %{}, fn char, acc ->
      Map.update(acc, char, 1, &(&1 + 1))
    end)

    Enum.find_index(s, fn char -> Map.get(freq, char) == 1 end) || -1
  end
end

IO.puts UniqueCharacter.first_uniq_char("leetcode")  # Output: 0 (Character 'l' is unique)
